controllers/velero.go

```diff
@@ -177,25 +170,25 @@ func (r *DPAReconciler) buildVeleroDeployment(veleroDeployment *appsv1.Deploymen
 		uploaderType = dpa.Spec.Configuration.NodeAgent.UploaderType
 	}
 
+	// TODO uploader type is passed to deployment container args, but not veleroContainer.Args. Is this a problem?
 	installDeployment := install.Deployment(veleroDeployment.Namespace,
 		install.WithResources(veleroResourceReqs),
 		install.WithImage(getVeleroImage(dpa)),
 		install.WithAnnotations(podAnnotations),
-		install.WithFeatures(dpa.Spec.Configuration.Velero.FeatureFlags),
-		install.WithUploaderType(uploaderType),
 		// last label overrides previous ones
 		install.WithLabels(veleroDeployment.Labels),
 		// use WithSecret false even if we have secret because we use a different VolumeMounts and EnvVars
 		// see: https://github.com/vmware-tanzu/velero/blob/ed5809b7fc22f3661eeef10bdcb63f0d74472b76/pkg/install/deployment.go#L223-L261
 		// our secrets are appended to containers/volumeMounts in credentials.AppendPluginSpecificSpecs function
 		install.WithSecret(false),
 		install.WithServiceAccountName(common.Velero),
+		// TODO remove
+		install.WithFeatures(dpa.Spec.Configuration.Velero.FeatureFlags),
+		install.WithUploaderType(uploaderType),
```

Comment:
this flag is never passed to velero container, because we override it afterwards. Should it be added?